Sources/AEPOptimize/Optimize+PublicAPI.swift

```diff
@@ -54,7 +55,17 @@ public extension Optimize {
                           source: EventSource.requestContent,
                           data: eventData)
 
-        MobileCore.dispatch(event: event)
+        MobileCore.dispatch(event: event, timeout: 10) { responseEvent in
+            guard let responseEvent = responseEvent else {
+                completion?(false, AEPError.callbackTimeout)
+                return
+            }
+            if let error = responseEvent.data?[OptimizeConstants.EventDataKeys.RESPONSE_ERROR] as? AEPError {
```

Comment:
nit: Can you check if you can use `getTypedData` here instead of casting?

`let error: AEPError =  responseEvent.getTypedData(for: OptimizeConstants.EventDataKeys.RESPONSE_ERROR)`